packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerBuilder.java

```diff
@@ -40,108 +40,143 @@ private static Paint setTextPaint(float size, Context context) {
         Path pointer = new Path();
         pointer.setFillType(Path.FillType.EVEN_ODD);
         float width = marker.getWidth();
-        float height = marker.getHeight() - bubblePointSize;
-        pointer.moveTo(width / 2f - bubblePointSize, height);
-        pointer.lineTo(width / 2f + bubblePointSize, height);
-        pointer.lineTo(width / 2f, height + bubblePointSize);
-        pointer.lineTo(width / 2f - bubblePointSize, height);
+        float height = marker.getHeight() - priceMarkerTailSize - shadowSize;
+        pointer.moveTo(width / 2f - priceMarkerTailSize, height);
+        pointer.lineTo(width / 2f + priceMarkerTailSize, height);
+        pointer.lineTo(width / 2f, height + priceMarkerTailSize);
+        pointer.lineTo(width / 2f - priceMarkerTailSize, height);
         pointer.close();
         return pointer;
     }
 
-    private Bitmap addBubbleMarkerText(String text) {
+    private Bitmap getPriceMarker(String text) {
         Rect rect = new Rect();
-        bubbleTextPaint.getTextBounds(text, 0, text.length(), rect);
+        priceMarkerTextStyle.getTextBounds(text, 0, text.length(), rect);
 
-        int padding = this.bubblePointSize * 2;
         int width = rect.width() + padding;
+        int height = rect.height() + padding;
+        int shadowWidth = width + shadowSize;
+        int shadowHeight = height + shadowSize;
+        Bitmap marker = Bitmap.createBitmap(shadowWidth, shadowHeight + priceMarkerTailSize, Bitmap.Config.ARGB_8888);
 
-        RectF bubble = new RectF(0, 0, width, rect.height() + padding);
+        RectF shadow = new RectF(0, 0, shadowWidth, shadowHeight);
+        RectF bubble = new RectF(shadowSize, shadowSize, width, height);
 
-        int height = rect.height() + padding + bubblePointSize;
-        Bitmap marker = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
         Canvas canvas = new Canvas(marker);
 
-        double density = Resources.getSystem().getDisplayMetrics().density;
-        int borderRadius = (int) (5 * density);
-        canvas.drawRoundRect(bubble, borderRadius, borderRadius, getShadowPaint());
-        canvas.drawRoundRect(bubble, borderRadius, borderRadius, getBackgroundColor());
-        canvas.drawPath(getBubblePoint(marker), getBackgroundColor());
+        int borderRadius = getBorderRadius(5);
+        int shadowBorderRadius = getBorderRadius(10);
+        canvas.drawRoundRect(shadow, shadowBorderRadius, shadowBorderRadius, getShadowPaint());
+        canvas.drawRoundRect(bubble, borderRadius, borderRadius, getMarkerPaint());
+        canvas.drawPath(getPriceMarkerTail(marker), getMarkerPaint());
+
+        float dx = getTextXOffset(width, rect);
+        float dy = getTextYOffset(height, rect);
+        canvas.drawText(text, dx, dy, priceMarkerTextStyle);
+        return marker;
+    }
+
+    private float getTextYOffset(float height, Rect rect) {
+        return (height / 2f) + (rect.height() / 2f) - rect.bottom;
```

Comment:
Can we rename the height argument to something more explicit? Is it MarkerHeight? CanvasHeight?

Reply:
Sure, just did it in a new commit